statement.go

```diff
@@ -1011,5 +1026,10 @@ func (stmt *Stmt) ociStmtExecute(iters C.ub4, mode C.ub4) error {
 		mode,                // The mode: https://docs.oracle.com/cd/E11882_01/appdev.112/e10646/oci17msc001.htm#LNOCI17163
 	)
 
+	if (result != C.OCI_SUCCESS && result != C.OCI_SUCCESS_WITH_INFO) && stmt.cacheKey != "" {
```

Comment:
Very minor performance improvement:
`if stmt.cacheKey != "" && result != C.OCI_SUCCESS && result != C.OCI_SUCCESS_WITH_INFO {`
Putting stmt.cacheKey first so does not need to check the rest in cases where cache is not enabled.
Also (  ) are not needed, which probably does not matter either way.